interface/ceed.c

```diff
@@ -616,22 +616,31 @@ int CeedInit(const char *resource, Ceed *ceed) {
   // LCOV_EXCL_STOP
   ierr = CeedRegisterAll(); CeedChk(ierr);
 
+  size_t stem_length;
+  for (stem_length=0; resource[stem_length]
+       && resource[stem_length] != ':'; stem_length++) {}
```

Comment:
This assumes the delinator for device information is a `:`

Reply:
That's what I was planning in my branch (I'll push it soon -- I'm running into some technical issues testing the HIP version to make sure it works, but the CUDA version is working to set the device to something other than 0).